jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -255,23 +259,20 @@ def vue_data_selected(self, event):
 
         # Also set the spectral min and max to default to the full range
         self.selected_subset = "Entire Spectrum"
+        # This is no longer needed for 1D but is preserved for now pending
+        # fixes to Cubeviz for multi-subregion subsets
         self._window = None
         self.spectral_min = selected_spec.spectral_axis[0].value
         self.spectral_max = selected_spec.spectral_axis[-1].value
         self.spectral_unit = str(selected_spec.spectral_axis.unit)
 
     def vue_list_subsets(self, event):
         """Populate the spectral subset selection dropdown"""
-        temp_subsets = self.app.get_subsets_from_viewer("spectrum-viewer",
-                                                        subset_type="spectral")
-        temp_dict = {}
-        # Attempt to filter out spatial subsets
-        for key, region in temp_subsets.items():
-            if type(region) == RectanglePixelRegion:
-                temp_dict[key] = region
-        self._spectral_subsets = temp_dict
-        self.spectral_subset_items = ["Entire Spectrum"] + sorted(temp_dict.keys())
+        self._spectral_subsets = self.app.get_subsets_from_viewer("spectrum-viewer",
+                                                                  subset_type="spectral")
+        self.spectral_subset_items = ["Entire Spectrum"] + sorted(self._spectral_subsets.keys())
 
+    '''
```

Comment:
Is this also pending cubeviz fix or is it dead code now?

Reply:
Good catch, I should have uncommented this for Cubeviz.

Reply:
Fixed, Cubeviz is happy (well, as happy as it was) again.